### PR TITLE
add awayUntil for members

### LINKED ISSUE #23: Member Status
As a member I want to set a state whether I am available or not.
If I am not available I want to define how long I am not available.
After the period ran out I want to get reminded to update my status.

### PR DESCRIPTION
**Pull request by Integer93**
closes #23

- Added awayUntil as Date field in member
  - used date because it is easy to use, easy to serialize and we don´t want to deal with timezones
- Added controller mapping to set it

### CODE REVIEW

**Comment by rainbowdashlabs** [comment]
Simply use timestamps in UTC. Thats the easy way. Not being able to mark yourself away for a few hours is kinda useless, because thats the most common use case.

File: src/main/java/io/github/opendme/server/controller/MemberController.java [hunk 1]
```diff
@@ -35,11 +37,19 @@ public Member create(@RequestBody @Valid MemberDto dto) {
...
     @ResponseStatus(HttpStatus.ACCEPTED)
     public void setStatus(@PathVariable Long memberId, @RequestBody Status status) {
         service.setMemberStatus(memberId, status);
         log.atInfo().log("Member status patched");
 
     }
+
+    @PatchMapping(value = "/member/{memberId}/awayUntil")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    public void setStatus(@PathVariable Long memberId, @RequestBody Date awayUntil) {
```
Comment (yannicklamprecht) [inline src/main/java/io/github/opendme/server/controller/MemberController.java:50]: Use OffsetDateTime instead to address @rainbowdashlabs remarks and to directly use modern date time types.

File: src/main/java/io/github/opendme/server/entity/Member.java [hunk 2]
```diff
@@ -28,6 +29,8 @@ public class Member {
     private String email;
     @Enumerated
     private Status status;
+    private Date awayUntil;
```
Comment (yannicklamprecht) [inline src/main/java/io/github/opendme/server/entity/Member.java:32]: private OffsetDateTime awayUntil;

**Review by yannicklamprecht — CHANGES_REQUESTED** [review]

**Comment by Integer93** [comment]
Date also contains the time. A date is basically 01.01.24 19:00:00. So its precise enough.
And its easy to serialize, every other DateTime variant was annoying.

I will try the Offset variant.

**Review by yannicklamprecht — APPROVED** [review]

**Comment by Integer93** [comment]
awayUntil ist now patchable with miliseconds since unix epoch.
See https://currentmillis.com/

**Pull request merged**